src/libs/SearchUtils.ts

```diff
@@ -76,9 +77,39 @@ function getShouldShowMerchant(data: OnyxTypes.SearchResults['data']): boolean {
     });
 }
 
+function doesAtleastOneExpenseBelongToAPastYear(data: TransactionListItemType[] | ReportListItemType[] | OnyxTypes.SearchResults['data']): boolean {
```

Comment:
Could you simplify this to:
- sort the available transactions by created
- take the oldest one
- check if its other than current year and return yes

We dont need to iterate over all of them, we just need to know if the oldest one is out of this year

Reply:
We can go ahead with using the `some` approach as that will be O(n) at worst

Reply:
Yea, was gonna say iterating over the list is O(n) (worst case) vs O(n log n) for sorting, so we should just iterate over the list with `some` as decided here